### PR TITLE
refactor!: split test_utils into public hassette.testing API

### LINKED ISSUE #1333: Separate public and internal testing infrastructure
## Description

Create `hassette.testing` as the sole supported app-author testing namespace and move Hassette-only test infrastructure outside the installed package.

The current `hassette.test_utils` package mixes two distinct products:

- A 16-name Tier 1 API documented for app authors.
- Dozens of Tier 2 fixtures, low-level harnesses, web mocks, factories, and helpers used only by Hassette's own test suite.

`__all__` documents the intended boundary, but it does not enforce one: Tier 2 names remain direct root-level attributes and imports, and `tests/unit/test_public_api_surface.py` explicitly guarantees some of them. The module name `test_utils` also reads as repository plumbing rather than a supported user API.

Keep the public testing API in the main Hassette distribution so it always versions with the framework. Do not create a separately versioned PyPI package. Internal-only support belongs under `tests/support` and should not ship in the wheel.

## Target Structure

```text
src/hassette/testing/        # shipped and supported
    __init__.py
    app_harness.py
    recording_api.py
    factories.py
    exceptions.py

tests/support/               # not shipped; no compatibility promise
    harness.py
    fixtures.py
    web.py
    websocket.py
    factories.py
```

Private implementation required by the public API may remain in underscore-prefixed modules under `hassette.testing`. General infrastructure used only by Hassette's own tests must not.

## Acceptance Criteria

- [ ] `hassette.testing` is the sole supported app-author testing namespace.
- [ ] Its root exports only intentionally supported public APIs.
- [ ] Hassette-only fixtures, harnesses, web mocks, websocket mocks, and factories move under `tests/support` or another non-installed test-support location.
- [ ] The built wheel does not contain Hassette-only test-support modules.
- [ ] Public testing code never imports from `tests/support`.
- [ ] Shared implementation required by public utilities is clearly private implementation, not a general internal fixture collection.
- [ ] Public docs and examples use `from hassette.testing import ...`.
- [ ] `hassette.test_utils` is removed rather than retained as an ambiguous internal or compatibility namespace.
- [ ] Hassette's own tests are migrated to public APIs or explicit test-support imports as appropriate.
- [ ] The hautomate consumer repository is migrated alongside the release.
- [ ] A built-wheel smoke test verifies every documented public testing import.
- [ ] Every current Tier 1 export is reviewed; `make_mock_hassette` remains public only if a concrete app-author job is documented.
- [ ] Pre-v1 migration notes identify removed and renamed imports.

## Non-Goals

- A separately published `hassette-testing` distribution
- Preserving undocumented Tier 2 root imports
- Moving useful public test utilities out of the Hassette release cadence
- Turning `tests/support` into another user-facing API

## Related

- #1286 tracks decomposition of the current internal harness. Its import-preservation assumptions should be revisited as part of this boundary change.
- #945 proposes an internal event-capture helper; if retained, its implementation belongs on the internal side of this boundary unless separately promoted.

### PR DESCRIPTION
**Pull request by NodeJSmith**
## Summary

Splits `hassette.test_utils` into two namespaces so app authors can tell supported test API
from internal framework plumbing: `hassette.testing` (ships in the wheel, `__all__`-gated
public surface) and `tests.support` (internal-only, outside `src/`, absent from the wheel).
`hassette.test_utils` no longer exists — no shim, no deprecation warning. See
`design/specs/108-testing-infra-split/design.md` for the full design and
`design/specs/108-testing-infra-split/known-issues.md` for a triaged-and-resolved finding
from implementation review.

### Public API (`hassette.testing`)

- `__all__` now contains exactly the Tier 1 symbols: `ApiCall`, `AppConfigurationError`,
  `AppTestHarness`, `DrainError`, `DrainFailure`, `DrainTimeout`, `EventCapture`,
  `HassetteHarness`, `RecordingApi`, `build_harness`, `create_call_service_event`,
  `create_state_change_event`, `dummy_cache`, `make_full_state_change_event`,
  `make_light_state_dict`, `make_sensor_state_dict`, `make_state_dict`,
  `make_switch_state_dict`, `make_test_config`, `make_typed_state`, `wait_for`.
- Modules a Tier 1 symbol transitively depends on ship in the wheel as private (`_`-prefixed)
  modules (`_harness.py`, `_simulation.py`, `_time_control.py`, `_sync_facade.py`, `_reset.py`,
  `_server.py`, `_ws_mocks.py`, `_factories.py`) — the boundary is namespace clarity via
  `__all__`, not wheel size.
- `dummy_cache` resolves lazily via module `__getattr__` so importing the rest of `hassette.testing`
  never pulls in `pytest` (an optional `[test]`-extra dependency) just because `fixtures.py`
  needs it.
- Tier 1 pytest fixtures (`dummy_cache`, `event_capture`) remain registerable by external
  consumers via `hassette.testing.fixtures`.

### Internal infrastructure (`tests.support`)

- Everything else — web-layer factories, seed scenario infrastructure, codegen-only modules,
  `mock_hassette.py`, `factories.py`, SQL/uvicorn test servers — moves to `tests/support/`,
  outside `src/` and absent from the wheel.
- Dependency direction is enforced one-way: `hassette.testing` never imports from
  `tests.support`; `tests/support/` may import from `hassette.testing`.
- `tools/check_module_boundaries.py` gained a `testing → tests.support` AST rule (#1333)
  enforcing that direction, since a plain grep can miss `import tests.support.*` forms.

### Verification

- `noxfile.py` gained a `wheel_smoke` session: builds the wheel, installs it into an isolated
  venv, confirms `from hassette.testing import AppTestHarness` succeeds and
  `import hassette.test_utils` raises `ModuleNotFoundError`.
- `tests/unit/test_public_api_surface.py` asserts `hassette.testing.__all__` matches the
  expected Tier 1 set.
- `tools/check_test_factories.py` gained a freshness check that was previously wired up but
  never actually called from `main()` — the check existed as dead code before this PR.

### Notable Changes

- Fixed `test_tier2_importable`, which was misleadingly asserting Tier 1 (not Tier 2)
  importability — renamed to `test_tier1_importable_from_testing` to match what it verifies.
- App init-failure tracebacks now show the app's own frame instead of the framework call site
  again (`get_short_traceback()` call sites restored to the negative-limit convention) —
  regressed earlier on this branch, caught and fixed during implementation review
  (see `known-issues.md` KI-001).

### Housekeeping

- Added `libcst>=1.5.0` as a dev dependency for the (throwaway) migration codemod tooling.
- Updated ~90 doc snippet and test files with mechanical import-path rewrites
  (`hassette.test_utils` → `hassette.testing` / `tests.support`).

## Breaking Changes

This PR removes `hassette.test_utils` entirely — no compatibility shim, no deprecation period.
External consumers importing from it must switch to the new locations.

BREAKING CHANGE: `hassette.test_utils` no longer exists. Public (Tier 1) symbols move to
`hassette.testing`; everything else moves to `tests.support` (internal-only, not shipped in
the wheel — not usable outside this repo).
#### Public API — update the import path only
- `ApiCall`, `AppConfigurationError`, `AppTestHarness`, `DrainError`, `DrainFailure`,
  `DrainTimeout`, `RecordingApi`, `create_call_service_event`, `create_state_change_event`,
  `dummy_cache`, `make_light_state_dict`, `make_sensor_state_dict`, `make_state_dict`,
  `make_switch_state_dict`, `make_test_config`, `make_typed_state` now import from
  `hassette.testing` instead of `hassette.test_utils`.
#### Public API — newly promoted to Tier 1 (previously internal-only)
- `EventCapture`, `HassetteHarness`, `build_harness`, `make_full_state_change_event`,
  `wait_for` are now part of the supported `hassette.testing` surface.
#### Removed from the public surface
- `make_mock_hassette` was previously in `hassette.test_utils.__all__`; it is now
  internal-only (`tests.support.mock_hassette`, not shipped in the wheel). External consumers
  using it should build their own `Hassette` stub, or open an issue if this needs to remain
  public.
#### Internal-only, no longer importable at all outside this repo
- Every symbol not listed above (web-layer factories, seed scenario helpers, SQL/uvicorn test
  servers, `FakeStateReader`, `create_app_manifest`, `make_addrinfo`, `patch_loop_getaddrinfo`,
  etc.) is no longer part of the wheel. It lived at `hassette.test_utils.*` before; it now lives
  at `tests.support.*` inside this repository only.

Closes #1333

### CODE REVIEW

**Comment by NodeJSmith** [comment]
@coderabbitai review

File: tools/check_module_boundaries.py [hunk 1]
```diff
@@ -234,30 +255,61 @@ def resolved_from_module(node: ast.ImportFrom, package: str | None) -> str | Non
...
+    Matches ``importlib.import_module("x")``, ``importlib.__import__("x")``, and bare
+    ``__import__("x")`` — the three ways to import a module by string name at runtime,
+    which bypass every static ``import``/``from`` form ``runtime_imports`` otherwise
+    looks for. Returns None for calls with no positional argument, a non-literal argument
+    (can't resolve statically — not flagged, since there's nothing to check), or a
+    different callee.
+    """
+    func = node.func
+    is_dynamic_callee = (isinstance(func, ast.Name) and func.id == "__import__") or (
+        isinstance(func, ast.Attribute) and func.attr in ("import_module", "__import__")
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tools/check_module_boundaries.py]: Fixed — added `dynamic_import_aliases()` to `tools/check_module_boundaries.py`, which walks the AST once per file to collect bound names from `from importlib import import_module` / `from importlib import __import__` (honoring `as` aliasing). `dynamic_import_target()` now checks `ast.Name` callees against that set alongside the existing `importlib.import_module(...)`/bare `__import__(...)` forms. Added a regression test (`test_testing_dynamic_import_bound_name_flagged`) reproducing the exact bypass.
